selfdrive/car/tesla/carcontroller.py

```diff
@@ -151,15 +178,67 @@ def update(self, sendcan, enabled, CS, frame, actuators, \
+        CS.DAS_info_frm = frame # 1.00 s interval
+        CS.DAS_status_frm = (frame + 10) % 100 # 0.50 s interval
+        CS.DAS_status2_frm = (frame + 35) % 100 # 0.50 s interval in between DAS_status
+        CS.DAS_bodyControls_frm = (frame + 40) % 100 # 0.50 s interval
+        CS.DAS_lanes_frm = (frame + 5) % 100 # 0.10 s interval 
+        CS.DAS_objects_frm = (frame + 2) % 100 # 0.03 s interval
+        CS.DAS_pscControl_frm = (frame + 3) % 100 # 0.04 s interval
+      if (CS.DAS_info_frm == frame):
+        can_sends.append(teslacan.create_DAS_info_msg(CS.DAS_info_msg))
+        CS.DAS_info_msg += 1
+        CS.DAS_info_msg = CS.DAS_info_msg % 10
+      if (CS.DAS_status_frm == frame):
+        can_sends.append(teslacan.create_DAS_status_msg(CS.DAS_status_idx))
+        CS.DAS_status_idx += 1
+        CS.DAS_status_idx = CS.DAS_status_idx % 16
+        CS.DAS_status_frm = (CS.DAS_status_frm + 50) % 100
+      if (CS.DAS_status2_frm == frame):
+        can_sends.append(teslacan.create_DAS_status2_msg(CS.DAS_status2_idx))
+        CS.DAS_status2_idx += 1
+        CS.DAS_status2_idx = CS.DAS_status2_idx % 16
+        CS.DAS_status2_frm = (CS.DAS_status2_frm + 50) % 100
+      if (CS.DAS_bodyControls_frm == frame):
+        can_sends.append(teslacan.create_DAS_bodyControls_msg(CS.DAS_bodyControls_idx))
+        CS.DAS_bodyControls_idx += 1
+        CS.DAS_bodyControls_idx = CS.DAS_bodyControls_idx % 16
+        CS.DAS_bodyControls_frm = (CS.DAS_bodyControls_frm + 50) % 100
+      if (CS.DAS_lanes_frm == frame):
+        can_sends.append(teslacan.create_DAS_lanes_msg(CS.DAS_lanes_idx))
+        CS.DAS_lanes_idx += 1
+        CS.DAS_lanes_idx = CS.DAS_lanes_idx % 16
+        CS.DAS_lanes_frm = (CS.DAS_lanes_frm + 10) % 100
+      if (CS.DAS_pscControl_frm == frame):
+        can_sends.append(teslacan.create_DAS_pscControl_msg(CS.DAS_pscControl_idx))
+        CS.DAS_pscControl_idx += 1
+        CS.DAS_pscControl_idx = CS.DAS_pscControl_idx % 16
+        CS.DAS_pscControl_frm = (CS.DAS_pscControl_frm + 4) % 100
+      if (CS.DAS_objects_frm == frame):
+        can_sends.append(teslacan.create_DAS_objects_msg(CS.DAS_objects_idx))
+        CS.DAS_objects_idx += 1
+        CS.DAS_objects_idx = CS.DAS_objects_idx % 16
+        CS.DAS_objects_frm = (CS.DAS_objects_frm + 3) % 100
+      # end of DAS emulation """
       idx = frame % 16 #(frame/send_step) % 16 
       can_sends.append(teslacan.create_steering_control(enable_steer_control, apply_angle, idx))
       can_sends.append(teslacan.create_epb_enable_signal(idx))
+      cruise_btn = None
+      if self.ACC.enable_adaptive_cruise:
+        cruise_btn = self.ACC.update_acc(enabled,CS,frame,actuators,pcm_speed)
+      if cruise_btn:
+        can_sends.append(teslacan.create_cruise_adjust_msg(cruise_btn,-1,CS.steering_wheel_stalk))
+      elif (turn_signal_needed > 0) and (frame % 3 ==0):
+        can_sends.append(teslacan.create_cruise_adjust_msg(-1,turn_signal_needed,CS.steering_wheel_stalk))
```

Comment:
Please change this ACC message from
"can_sends.append(foo)"
to
"can_sends.insert(0, foo)"

So that the cruise messages are the first ones sent each cycle. Since the spam cruise control messages are racing against the real cruise stalk, I it's important that they are sent as soon as possible.